core/src/avm2/value.rs

```diff
@@ -1315,21 +1315,26 @@ impl<'gc> Value<'gc> {
         other: &Value<'gc>,
         activation: &mut Activation<'_, 'gc>,
     ) -> Result<Option<bool>, Error<'gc>> {
-        let prim_self = self.coerce_to_primitive(Some(Hint::Number), activation)?;
```

Comment:
I like 
```rust
        if let (Value::Integer(a), Value::Integer(b)) = (self, other) {
            return Ok(Some(a < b));
        }
```

Reply:
I think the current match syntax is more readable here, and it allows for easily adding more cases if/when we want them